patches/server/0749-Implement-CommandChatEvent.patch

```diff
@@ -0,0 +1,279 @@
++            }
++            net.kyori.adventure.text.Component[] arguments = new net.kyori.adventure.text.Component[]{ displayName, adventureMessage };
++
++            java.util.Set<net.kyori.adventure.audience.Audience> viewers = new java.util.HashSet<>(1);
++            viewers.add(this.server.console);
++            io.papermc.paper.event.command.CommandChatEvent event = new io.papermc.paper.event.command.CommandChatEvent(this.getBukkitSender(), viewers, io.papermc.paper.chat.CommandChatRenderer.adminRenderer(), arguments, originalArguments, net.kyori.adventure.audience.MessageType.SYSTEM, io.papermc.paper.event.command.CommandChatEvent.Type.ADMIN_NOTIFICATION);
++            if (event.callEvent()) {
++                for (net.kyori.adventure.audience.Audience viewer : event.viewers()) {
++                    viewer.sendMessage(net.kyori.adventure.identity.Identity.nil(), event.renderer().render(this.getBukkitSender(), new net.kyori.adventure.text.Component[]{ event.argument(0), event.argument(1) }, viewer), event.messageType());
++                }
++            }
++            // Paper end
++            // this.server.sendMessage(ichatmutablecomponent, Util.NIL_UUID); // Paper - send to audiences
+         }
+ 
+     }
+diff --git a/src/main/java/net/minecraft/server/commands/EmoteCommands.java b/src/main/java/net/minecraft/server/commands/EmoteCommands.java
+index 5658e6653f782895d264f9f5e7dc92f9e2f364b5..9583c7d081cfacbe7b2caac14e8936a737527329 100644
+--- a/src/main/java/net/minecraft/server/commands/EmoteCommands.java
++++ b/src/main/java/net/minecraft/server/commands/EmoteCommands.java
+@@ -19,29 +19,98 @@ public class EmoteCommands {
+             String string = StringArgumentType.getString(context, "action");
+             Entity entity = context.getSource().getEntity();
+             MinecraftServer minecraftServer = context.getSource().getServer();
++            // Paper start
++            io.papermc.paper.adventure.LazyChatAudienceSet viewers = new io.papermc.paper.adventure.LazyChatAudienceSet(context.getSource().getServer());
++            net.kyori.adventure.text.Component displayName = io.papermc.paper.adventure.PaperAdventure.asAdventure(context.getSource().getDisplayName());
++            // Paper end
+             if (entity != null) {
++                net.kyori.adventure.identity.Identity identity = net.kyori.adventure.identity.Identity.identity(entity.getUUID()); // Paper
+                 if (entity instanceof ServerPlayer) {
+                     ServerPlayer serverPlayer = (ServerPlayer)entity;
+                     serverPlayer.getTextFilter().processStreamMessage(string).thenAcceptAsync((message) -> {
++                        /* // Paper - move down and edit to filter after event
+                         String string = message.getFiltered();
+                         Component component = string.isEmpty() ? null : createMessage(context, string);
++                         */
```

Comment:
I think we stopped using, (or are trying to stop) block comments, because if upstream changes something inside it, it doesn't always conflict which makes the change go ignored until something breaks.

Reply:
I realize that the variables aren't actually used but also they don't affect anything, should I uncomment them out?